### PR TITLE
Add a Color Scheme setting for Menus-Header

### LINKED ISSUE #2565: [Header] Add a Color Scheme setting for Menus
The idea would be to add a Color Scheme setting for Menus vs Header.
- Figma file is available [here](https://www.figma.com/file/LlopBqbeLND3IQQvYMjVbn/%F0%9F%9A%A7-Navigation-UX-%E2%80%93-Dawn-2023?node-id=2712%3A109466&t=OCQ8QopmMK3YHkqW-1).
- Both In-menu color scheme and Header color scheme settings should be gathered under a "Color" heading. That colors group of settings should be put above the Spacing setting.
- Only 2 colors are used in menus : 1 for the menus background, 1 for the links / icons / transparent background (we can see this on 3rd level links in dropdown and at the bottom of the drawer when social icons are used)

### PR DESCRIPTION
**Pull request by lougoncharenko**
### PR Summary: 
Added the ability to use a different color scheme for header and menu.

### Why are these changes introduced?
Added a second color scheme for navigation, so that the header and navigation can use different color schemes.
Fixes #2565 

### What approach did you take?
Kept original `color_scheme` setting for header with updated label. 
And added new setting `menu_color_scheme`
Added "color" header to header section schema instead of global translation

### Visual impact on existing themes

Not applicable.

### Testing steps/scenarios

- [ ]  Changed color scheme for menu to another scheme
- [ ] Ensured that the drop down can render in a different scheme than the first of the header
- [ ] Test all menu types (dropdown, drawer, etc)
- [ ] Test the color change with the social links
- [ ] Test with gradient
- [ ] Test with sticky header on/off
- [ ] Verify the search drawer inherits the header scheme
- [ ] verify cart notification inherits the header scheme

### Demo links

- [Editor](https://os2-demo.myshopify.com/admin/themes/139925651478/editor)

### Checklist
- [ ] Added PR summary for [release notes](https://themes.shopify.com/themes/dawn/styles/default#ReleaseNotes)
- [ ] Requested review from UX (Only for changes that are affecting the experience or perceivable visual details)
- [ ] Created a ticket for the [help.shopify.com](https://help.shopify.com) documentation team about updates to theme settings. (Internal-only task)
- [ ] Followed [theme code principles](https://github.com/Shopify/dawn/blob/main/.github/CONTRIBUTING.md#theme-code-principles)
- [ ] Linted with [Theme Check](https://github.com/Shopify/theme-check)
- [ ] Tested on [mobile](https://shopify.dev/themes/store/requirements#mobile-browser-requirements)
- [ ] Tested on [multiple browsers](https://shopify.dev/themes/store/requirements#desktop-browser-requirements)
- [ ] Tested for [accessibility](https://shopify.dev/themes/best-practices/accessibility)

### CODE REVIEW

File: snippets/header-drawer.liquid [hunk 1]
```diff
@@ -5,7 +5,7 @@
   {% render 'header-drawer' %}
 {% endcomment %}
 
-<header-drawer data-breakpoint="{% if section.settings.menu_type_desktop == 'drawer' %}desktop{% else %}tablet{% endif %}">
+<header-drawer  data-breakpoint="{% if section.settings.menu_type_desktop == 'drawer' %}desktop{% else %}tablet{% endif %}">
```
Comment (eugenekasimov) [inline snippets/header-drawer.liquid:8]:
```suggestion
<header-drawer  data-breakpoint="{% if section.settings.menu_type_desktop == 'drawer' %}desktop{% else %}tablet{% endif %}">
```

Comment (eugenekasimov) [inline snippets/header-drawer.liquid:8]:
```suggestion
<header-drawer data-breakpoint="{% if section.settings.menu_type_desktop == 'drawer' %}desktop{% else %}tablet{% endif %}">
```

File: snippets/header-drawer.liquid [hunk 2]
```diff
@@ -16,7 +16,7 @@
         {% render 'icon-close' %}
       </span>
     </summary>
-    <div id="menu-drawer" class="gradient menu-drawer motion-reduce">
+    <div id="menu-drawer"  class="gradient menu-drawer motion-reduce color-{{ section.settings.menu_color_scheme }}">
```
Comment (eugenekasimov) [inline snippets/header-drawer.liquid:19]:
```suggestion
    <div id="menu-drawer" class="gradient menu-drawer motion-reduce color-{{ section.settings.menu_color_scheme }}">
```

Comment (eugenekasimov) [inline snippets/header-drawer.liquid:19]:
```suggestion
    <div id="menu-drawer" class="gradient menu-drawer motion-reduce color-{{ section.settings.menu_color_scheme }}">
```

File: snippets/header-dropdown-menu.liquid [hunk 3]
```diff
@@ -5,7 +5,7 @@
   {% render 'header-dropdown-menu' %}
 {% endcomment %}
 
-<nav class="header__inline-menu">
+<nav class="header__inline-menu " >
```
Comment (eugenekasimov) [inline snippets/header-dropdown-menu.liquid:8]:
```suggestion
<nav class="header__inline-menu">
```

**Review by eugenekasimov — COMMENTED** [review]
Hey Louisa, everything looks good to me, just left minor suggestions. It looks like you accidentally added whitespace.

File: snippets/header-drawer.liquid (hunk 2 above)
Comment (lougoncharenko) [reply snippets/header-drawer.liquid:19]: removed the white space here as well!

File: snippets/header-dropdown-menu.liquid (hunk 3 above)
Comment (lougoncharenko) [reply snippets/header-dropdown-menu.liquid:8]: Removed the space here

File: locales/en.default.schema.json [hunk 4]
```diff
@@ -1063,6 +1063,15 @@
         "show_line_separator": {
           "label": "Show separator line"
         },
+        "header__1": {
+          "content": "Color"
+        },
+        "header_color_scheme": {
+          "label": "Header color scheme"
```
Comment (kmeleta) [inline locales/en.default.schema.json:1070]: @YoannJailin I don't feel strongly at all, but just wondering if we don't want to keep the original "Color scheme" label for the header's color scheme setting. I could certainly see this adding some clarity for users, but the setting is within the "Header" section as it is, so I somewhat feel like we don't need the additional qualifier.

We only have a couple other instances of multiple color schemes in one section to pull precedent from but they just use "Color scheme" to represent the section's color scheme.

File: snippets/header-dropdown-menu.liquid [hunk 5]
```diff
@@ -5,7 +5,7 @@
   {% render 'header-dropdown-menu' %}
 {% endcomment %}
 
-<nav class="header__inline-menu">
+<nav class="header__inline-menu ">
```
Comment (kmeleta) [inline snippets/header-dropdown-menu.liquid:8]: Not blocking but if we need to make other changes let's clean up this last bit of whitespace.

File: sections/header.liquid [hunk 6]
```diff
@@ -164,7 +164,7 @@
       assign localization_forms = true
     endif
   -%}
-  <header class="header header--{{ section.settings.logo_position }} header--mobile-{{ section.settings.mobile_logo_position }} page-width{% if section.settings.menu_type_desktop == 'drawer' %} drawer-menu{% endif %}{% if section.settings.menu != blank %} header--has-menu{% endif %}{% if has_app_block %} header--has-app{% endif %}{% if social_links %} header--has-social{% endif %}{% if shop.customer_accounts_enabled %} header--has-account{% endif %}{% if localization_forms %} header--has-localizations{% endif %}">
+  <header class="header color-{{ section.settings.color_scheme }} header--{{ section.settings.logo_position }} header--mobile-{{ section.settings.mobile_logo_position }} page-width{% if section.settings.menu_type_desktop == 'drawer' %} drawer-menu{% endif %}{% if section.settings.menu != blank %} header--has-menu{% endif %}{% if has_app_block %} header--has-app{% endif %}{% if social_links %} header--has-social{% endif %}{% if shop.customer_accounts_enabled %} header--has-account{% endif %}{% if localization_forms %} header--has-localizations{% endif %}">
```
Comment (kmeleta) [inline sections/header.liquid:167]: I actually think this is breaking the gradient functionality for the header. The color schemes still apply but will only display the fallback solid color. 

We already have a `color-` and `gradient` class the the parent element on line 154 and I think having it here too is effectively overriding the gradient class behavior. I suspect if we just remove this instance of the class all will work as expected.

Comment (lougoncharenko) [reply sections/header.liquid:167]: I removed the code from line 167 and kept the one in line 154 and the color scheme is still working for both header and menu

File: locales/en.default.schema.json (hunk 4 above)
Comment (YoannJailin) [reply locales/en.default.schema.json:1070]: Yes i agree we can replace "Header color scheme" by "Color scheme". I would keep the second one named "Menu color scheme". @lougoncharenko @kmeleta

**Review by YoannJailin — APPROVED** [review]

File: locales/en.default.schema.json (hunk 4 above)
Comment (lougoncharenko) [reply locales/en.default.schema.json]: Changed it back to the original translation and removed the header_color_scheme translation

**Review by eugenekasimov — COMMENTED** [review]
Looks good to me. I'll approve once translations are done 👍

**Pull request merged**